web/src/scheduled_messages.js

```diff
@@ -0,0 +1,158 @@
+import $ from "jquery";
+
+import render_scheduled_message from "../templates/scheduled_message.hbs";
+import render_scheduled_messages_overlay from "../templates/scheduled_messages_overlay.hbs";
+
+import * as blueslip from "./blueslip";
+import * as browser_history from "./browser_history";
+import * as channel from "./channel";
+import * as compose from "./compose";
+import * as compose_actions from "./compose_actions";
+import * as compose_fade from "./compose_fade";
+import * as compose_ui from "./compose_ui";
+import * as loading from "./loading";
+import * as narrow from "./narrow";
+import * as overlays from "./overlays";
+import * as people from "./people";
+import * as stream_color from "./stream_color";
+import * as stream_data from "./stream_data";
+import * as timerender from "./timerender";
+
+// This is only updated when user opens the scheduled messages overlay.
+let scheduled_messages_data = [];
+
+function hide_loading_indicator() {
+    loading.destroy_indicator($("#scheduled_messages_overlay .loading-indicator"), {
+        abs_positioned: true,
+    });
+    $(".scheduled-messages-loading").hide();
+}
+
+export function edit_scheduled_message(scheduled_msg_id) {
+    const scheduled_msg = scheduled_messages_data.find(
+        (msg) => msg.message_id === scheduled_msg_id,
+    );
+
+    let compose_args;
+
+    if (scheduled_msg.type === "stream") {
+        compose_args = {
+            type: "stream",
+            stream: scheduled_msg.stream_name,
+            topic: scheduled_msg.topic,
+            content: scheduled_msg.content,
+        };
+    } else {
+        const recipient_emails = [];
+        if (scheduled_msg.to) {
+            for (const recipient_id of scheduled_msg.to) {
+                recipient_emails.push(people.get_by_user_id(recipient_id).email);
+            }
+        }
+        compose_args = {
+            type: scheduled_msg.type,
+            private_message_recipient: recipient_emails.join(","),
+            content: scheduled_msg.content,
+        };
```

Comment:
Hmm, how does this store the ID of the scheduled message object that should be edited?

Reply:
Added to TODO list at the start of the PR.